Helper/PaymentResponseHandler.php

```diff
@@ -341,4 +357,65 @@ private function isValidMerchantReference(array $paymentsDetailsResponse, OrderI
 
         return true;
     }
+
+    // Method to check for existing Gift Card payments
+    private function hasActiveGiftCardPayments($merchantReference, $order)
```

Comment:
This method not only checks if the payment "has active giftcard payments" but it also initializes the API client and attempts to cancel the order.
For proper separation of concerns, I would recommend implementing separate methods to check for used giftcards, and to make cancel calls (preferably using and reusable helper method used elsewhere)